ambry-quota/src/main/java/com/github/ambry/quota/MySqlStorageUsageRefresher.java

```diff
@@ -0,0 +1,470 @@
+   */
+  private void initializeContainerStorageUsageMonthlyBase() {
+    // First try to get the monthly base storage usage from backup
+    try {
+      if (backupFileManager != null) {
+        containerStorageUsageMonthlyBase = backupFileManager.getBackupFileContent(currentMonth);
+      }
+      if (containerStorageUsageMonthlyBase != null) {
+        return;
+      }
+    } catch (IOException e) {
+      logger.error("Failed to get container monthly usage for " + currentMonth + " from backup", e);
+    }
+
+    try {
+      // If we are here, then loading monthly base from backup file failed. We have to fetch it from database.
+      containerStorageUsageMonthlyBase =
+          accountStatsMySqlStore.queryMonthlyAggregatedStats(clusterMapConfig.clusterMapClusterName);
+      // If the monthly base is indeed for this month, then try to persist it in the backup file.
+      // There is a chance that the database has a snapshot from last month since the aggregation task is executed
+      // every few minutes(maybe hours). Before the first aggregation task of this month is executed, the database
+      // would have the snapshot of last month.
+      if (currentMonth.equals(accountStatsMySqlStore.queryRecordedMonth(clusterMapConfig.clusterMapClusterName))) {
+        tryPersistMonthlyUsage();
+      }
+    } catch (Exception e) {
+      throw new IllegalStateException("Unable to fetch monthly storage usage from mysql", e);
+    }
+  }
+
+  /**
+   * Try to persist the monthly base usage in the backup file and ignore any exceptions.
+   */
+  private void tryPersistMonthlyUsage() {
+    if (backupFileManager != null) {
+      try {
+        backupFileManager.persistentBackupFile(currentMonth, containerStorageUsageMonthlyBase);
+      } catch (IOException e) {
+        // Error already been logged from backup file manager.
+      }
+    }
+  }
+
+  /**
+   * Fetch the container monthly storage usage from mysql and update the in memory cache. If somehow it fails, we will
+   * retry for several times before giving up.
+   */
+  private void fetchMonthlyStorageUsageAndMaybeRetry() {
+    boolean shouldRetry = false;
+    try {
+      String monthValue = accountStatsMySqlStore.queryRecordedMonth(clusterMapConfig.clusterMapClusterName);
+      if (monthValue.equals(currentMonth)) {
+        containerStorageUsageMonthlyBase =
+            accountStatsMySqlStore.queryMonthlyAggregatedStats(clusterMapConfig.clusterMapClusterName);
+      } else {
+        shouldRetry = true;
+      }
+    } catch (Exception e) {
+      logger.error("Failed to refresh monthly storage usage", e);
+      shouldRetry = true;
+    }
+
+    if (shouldRetry) {
+      if (retries >= config.mysqlStoreRetryMaxCount) {
+        logger.error("Failed to refresh monthly storage usage after " + retries + " retries, will skip for month: "
+            + currentMonth);
+        retries = 0;
+        scheduleStorageUsageMonthlyBaseFetcher();
```

Comment:
So in this case is the base value the last month's value? If yes, then can customers be penalized for our failure to get monthly base. 
I am worried about the following scenario:
example accountId: A, containerId: B, container quota: 12G
current date: Dec 2
Base Value Nov = 85G
Base Value Dec = 90G (this could not be fetched due to some reason), so containerStorageUsageMonthlyBase["A"]["B"] = 85G
Current Usage Value for container = 100G
Whats our response in this case?